OpenNefia.Content/UI/Element/UiMessageWindowBacking.cs

```diff
@@ -0,0 +1,97 @@
+﻿using OpenNefia.Core.IoC;
+using OpenNefia.Core.Rendering;
+using OpenNefia.Core.UI.Element;
+using OpenNefia.Content.Prototypes;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using OpenNefia.Core.Prototypes;
+using static OpenNefia.Core.Rendering.AssetInstance;
+using OpenNefia.Core.Maths;
+
+namespace OpenNefia.Content.UI.Element
+{
+    public class UiMessageWindowBacking : BaseDrawable
+    {
+        public const int MessageBoxWidth = 192;
+        public const int MessageBoxHeight = 72;
+        public enum MessageBackingType
+        {
+            Default,
+            Expanded
+        }
+
+        private IAssetInstance MessageWindowAsset = default!;
+        private Love.SpriteBatch Batch = default!;
+        private Love.SpriteBatch? SideBatch = default!;
+        private Love.SpriteBatch? CornerBatch = default!;
+        private MessageBackingType Type;
+        public UiMessageWindowBacking(MessageBackingType type = MessageBackingType.Default)
+        {
+            Type = type;
+        }
+
+        private int GetSideCount()
+        {
+            return Height / MessageBoxWidth;
+        }
+
+        public override void SetSize(int width, int height)
+        {
+            base.SetSize(width, height);
+            var parts = new List<AssetBatchPart>();
+
+            MessageWindowAsset = Assets.Get(Protos.Asset.MessageWindow);
+            switch (Type)
+            {
+                default:
+                    for (int x = 0; x < Width / MessageBoxWidth; x++)
+                    {
+                        parts.Add(new AssetBatchPart("topBar", (x * MessageBoxWidth), 0));
```

Comment:
Maybe these could be `const string` somewhere.

Reply:
changed in 9eb449f